activerecord/lib/active_record/disable_joins_association_relation.rb

```diff
@@ -30,7 +30,11 @@ def load
         record[key]
       end
 
-      records = ids.flat_map { |id| records_by_id[id.to_i] }
```

Comment:
I did a little local testing and I think we can remove the `to_i` here instead of doing the attribute type and casting.

Reply:
Ohhh interesting, I presumed it needed to be there and didn't think to try removing it. But simply dropping the `#to_i` does seem to keep the tests passing, and works in our app. :+1:

I'll modify the PR.

Reply:
I've amended the commit to remove the cast, instead.